packages/turbo/ios/RNSession.swift

```diff
@@ -116,6 +116,15 @@ extension RNSession: SessionDelegate {
   func sessionDidFinishFormSubmission(_ session: Session) {
     visitableView?.didFinishFormSubmission()
   }
+
+  func session(_ session: Session, decidePolicyFor navigationAction: WKNavigationAction) -> WebViewPolicyManager.Decision {
+    guard let url = navigationAction.request.url else {
+      return .allow
+    }
+    // regardless of the return value here nothing happens, so we have to manually open external URL
+    visitableView?.didOpenExternalUrl(url: url)
+    return .allow
+  }
```

Comment:
This has introduced a new behavior in our app. This seems to be called for on-page `<iframe src="..">` elements.

As an example, in our app we use Google Tag Manager. They have an initialization script that includes an `<iframe src="https://www.googletagmanager.com/ns.html?...">`. On every page view in the app, we see that `didOpenExternalUrl` is called with the Google Tag Manager URL with no way to differentiate if it was a user-decided navigation action or an on-page iframe.

Reply:
Hotwire Native has an extension that provides some additional on if the navigationAction happened in the main frame etc: https://github.com/hotwired/hotwire-native-ios/blob/f18b87938f55cbe8976e96702b5512fa9afcd674/Source/Turbo/Navigator/Extensions/WKNavigationAction%2BUtils.swift#L1C1-L12C1

I think there's a few things going on here:

1. The previous `Session#openExternalURL` that we relied on to trigger the `didOpenExternalUrl` on the RN-side was changed in Hotwire Native iOS to support the cross-origin redirects.
2. We have not yet fully supported the change
3. The current implementation in this PR is to naive and triggers even for iframes etc.

Reply:
Opened a PR #225 that addresses this.